### PR TITLE
Sign apk for Android release builds

### PR DESCRIPTION
**Pull request by samartnik**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by samartnik** [comment]
This is follow up for https://github.com/brave/brave-browser/issues/7383
This PR allows to set apk names in config only

File: build/android/sign_apk.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import sys
+import subprocess
+import tempfile
+
+if len(sys.argv) != 8:
+  print("Incorrect number of parameters")
+  print("Usage: " + sys.argv[0] + " <zipalign_path>, <apksigner_path>, <unsigned_apk_path>, <key_path>, <key_passwd>, <prvt_key_passwd>, <key_name>")
+  sys.exit(1)
+
+zipalign_path = sys.argv[1]
```
Comment (bridiver) [inline build/android/sign_apk.py]: this is ok if this is definitely temporary, but normally you should use `argparse` for the arguments

File: build/android/sign_apk.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+apksigner_path = sys.argv[2]
+unsigned_apk_path = sys.argv[3]
+key_path = sys.argv[4]
+key_passwd = sys.argv[5]
+prvt_key_passwd = sys.argv[6]
+key_name = sys.argv[7]
+
+with tempfile.NamedTemporaryFile() as staging_file:
```
Comment (bridiver) [inline build/android/sign_apk.py]: I think this stuff should be wrapped in a more typical `Main` method setup to make sure we're getting errors back to gn correctly
```
def Main():
  ...

if __name__ == '__main__':
  sys.exit(Main())
```

File: build/config.gni [hunk 3]
```diff
@@ -26,6 +26,10 @@ declare_args() {
   build_omaha = false
   tag_ap =""
   target_apk_base=""
+  brave_android_keystore_path = ""
```
Comment (bridiver) [inline build/config.gni]: config args already exist for most of these
android_keystore_path, android_keystore_name and android_keystore_password

Comment (samartnik) [reply build/config.gni]: If we specify `android_keystore_path` if will go through `package_apk` with errors. I specifically added `brave_` to not overlap with them

File: build/android/sign_apk.py (hunk 1 above)
Comment (samartnik) [reply build/android/sign_apk.py]: Changed that

File: build/android/sign_apk.py (hunk 2 above)
Comment (samartnik) [reply build/android/sign_apk.py]: Done

**Pull request merged**